tests/integrationv2/test_signature_algorithms.py

```diff
@@ -107,18 +106,26 @@ def test_s2n_server_signature_algorithms(managed_process, cipher, provider, othe
         results.assert_success()
         assert to_bytes("Actual protocol version: {}".format(
             expected_version)) in results.stdout
-        assert signature_marker(Provider.ServerMode,
-                                signature) in results.stdout
-        assert (signature_marker(Provider.ClientMode, signature)
-                in results.stdout) == client_auth
+        # TLS1.1 always negotiates to RSA+MD5_SHA1 signature algorithm.
+        if protocol < Protocols.TLS12:
+            assert signature_marker(
+                Provider.ServerMode, Signatures.MD5_SHA1
+            ) in results.stdout
+            assert (signature_marker(Provider.ClientMode, Signatures.MD5_SHA1)
+                    in results.stdout) == client_auth
+        else:
+            assert signature_marker(Provider.ServerMode,
+                                    signature) in results.stdout
+            assert (signature_marker(Provider.ClientMode, signature)
+                    in results.stdout) == client_auth
```

Comment:
I think you can make this cleaner / more readable. The two branches of your if/else are very similar. 

Can you calculate the expected_signature once for this test, outside of this loop? Or since you do it in two tests, maybe write a little helper method to calculate it? And then you don't need the if/else inside of the loop.